app/controllers/main/routes.py

```diff
@@ -47,50 +49,134 @@ def events(selectedTerm=None):
         currentTime = currentTime,
         user = g.current_user)
 
-@main_bp.route('/profile/<username>', methods = ['GET'])
-def profilePage(username):
-    if not g.current_user.isCeltsAdmin and not g.current_user.isCeltsStudentStaff and g.current_user.username != username:
-        return "Access Denied", 403
+@main_bp.route('/profile/<username>', methods=['GET'])
+def viewVolunteersProfile(username):
+    """
+    This function displays the information of a volunteer to the user
+    """
+    try:
+        User.get(User.username == username)
+    except Exception as e:
+        return "User does not exist", 404
+    if (g.current_user.username == username) or g.current_user.isAdmin:
```

Comment:
since you have already gotten the user from the database in line 58 (you've done the query anyway), I would just use it here instead of username, like `g.current_user == user`

Reply:
Turned line 61 into `if (g.current_user == username) or g.current_user.isAdmin:` and it works fine. I am not sure if that's what you meant by this comment, though.